launcher/settingsView/csettingsview_moc.cpp

```diff
@@ -110,6 +106,66 @@ void CSettingsView::loadSettings()
 	ui->comboBoxAutoSave->setCurrentIndex(settings["general"]["saveFrequency"].Integer() > 0 ? 1 : 0);
 }
 
+void CSettingsView::fillValidResolutions(bool isExtraResolutionsModEnabled)
+{
+	this->isExtraResolutionsModEnabled = isExtraResolutionsModEnabled;
+	fillValidResolutionsForScreen(ui->comboBoxDisplayIndex->isVisible() ? ui->comboBoxDisplayIndex->currentIndex() : 0);
+}
+
+void CSettingsView::fillValidResolutionsForScreen(int screenIndex)
+{
+	ui->comboBoxResolution->blockSignals(true); // avoid saving wrong resolution after adding first item from the list
+	ui->comboBoxResolution->clear();
+
+	// TODO: read available resolutions from all mods
+	QVariantList resolutions;
+	if(isExtraResolutionsModEnabled)
+	{
+		const auto extrasResolutionsPath = settings["launcher"]["extraResolutionsModPath"].String().c_str();
+		const auto extrasResolutionsJson = JsonUtils::JsonFromFile(CLauncherDirs::get().modsPath() + extrasResolutionsPath);
+		resolutions = extrasResolutionsJson.toMap().value(QLatin1String{"GUISettings"}).toList();
+	}
+	if(resolutions.isEmpty())
+	{
+		ui->comboBoxResolution->blockSignals(false);
+		ui->comboBoxResolution->addItem(resolutionToString({800, 600}));
+		return;
+	}
+
+	const auto screens = qGuiApp->screens();
+	const auto currentScreen = screenIndex < screens.size() ? screens[screenIndex] : qGuiApp->primaryScreen();
+	const auto screenSize = currentScreen->size();
```

Comment:
Looks like this variable can be also wrapped with
#ifndef VCMI_IOS
?

Reply:
All the 3 variables can be, but I prefer to keep less ifdefs - I believe in this case compiler would just throw away unused variables.

Reply:
agree, let's try to keep less platform dependent logic, especially in launcher